tests/integration/devfile/cmd_devfile_delete_test.go

```diff
@@ -34,19 +32,23 @@ var _ = Describe("odo devfile delete command tests", func() {
 	})
 
 	Context("when devfile delete command is executed", func() {
-		It("should not throw an error with an existing namespace when no component exists", func() {
-			helper.CmdShouldPass("odo", "create", "nodejs", "--project", commonVar.Project, componentName)
+		JustBeforeEach(func() {
 			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
-			helper.CmdShouldPass("odo", "delete", "--project", commonVar.Project, "-f")
+			helper.CmdShouldPass("odo", "create", "nodejs", componentName, "--project", commonVar.Project)
+		})
+
+		// Why do we have this test case? `It` part doesn't make sense
+		It("should not throw an error with an existing namespace when no component exists", func() {
+			helper.CmdShouldPass("odo", "delete", componentName, "--project", commonVar.Project, "-f")
+		})
```

Comment:
Why do we have this test? What is it trying to test?

Reply:
I _think_ the idea is to not fail if a component with the name `componentName` doesn't exist in the namespace. But I think it should fail in such a scenario.

@mik-dass any thoughts?

Reply:
The test is checking that when a component doesn't on the cluster, odo delete doesn't fail https://github.com/openshift/odo/blob/c0be940e1df15ca1ebf410de962932b327d9d5a7/pkg/devfile/adapters/kubernetes/component/adapter.go#L592-L595

> But I think it should fail in such a scenario. 

It doesn't fail since even if the component doesn't exist we still delete the linking information from the env file. Also if `-all` flag is provided, we delete the env file and other configuration too, even when the component doesn't exist.

Reply:
Oh okay, got it. It's the most basic scenario. :facepalm: I was confused because of the  `It` description. It would have made more sense to me if it just said that "should not throw an error when a component is not pushed".